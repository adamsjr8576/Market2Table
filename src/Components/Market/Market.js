import React from 'react';

const Market = ({ id, marketName }) => {
  return (
    <article>
      <h1>{marketName}</h1>
    </article>
  )
}

export default Market;
