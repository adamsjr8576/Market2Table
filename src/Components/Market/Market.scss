.market-article {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 500px;
  padding: 12px;
  border: 3px solid black;
  background-color: #FFE7BF;
  border-radius: 30px;
  text-align: center;
}

.market-article-name {
  font-size: 1.25em;
  text-align: center;
  margin: 0px 5px;
}

.chili-icon {
  height: 30px;
  width: 30px;
  margin-right: 20px;
  cursor: pointer;
}

.favorites-button {
  border: 0;
  background-color: transparent;
  outline: 0;
}

.favorites-button:hover,
.favorites-button:focus {
  transform: scale(1.2);
}

.market-link-to-page {
  text-decoration: none;
  color: black;
  border: 1px solid black;
  margin-left: 10px;
  padding: 10px;
  border-radius: 30px;
  background-color: white;
}

.market-link-to-page:hover,
.market-link-to-page:focus {
  transform: scale(1.02);
}
