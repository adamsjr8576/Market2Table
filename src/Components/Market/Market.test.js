import React from 'react';
import {shallow} from 'enzyme';
import { Market, mapDispatchToProps } from './Market';
import { toggleFavorite, controlFavorites } from '../../actions/index.js';

describe('Market', () => {
  describe('Market Component', () => {
    let mockToggleFavorite;
    let mockControlFavorites;
    let mockMarket;
    let mockImage;

    beforeEach(() => {
      mockToggleFavorite = jest.fn();
      mockControlFavorites = jest.fn();
      mockMarket = {
        id: '1234',
        marketname: 'Golden Farmers Markets',
        favorite: true
      }
      mockImage = './redChili.png';
    });

    it('should match the snapshot with all the correct data passed in', () => {
      const wrapper = shallow(
        <Market
          market={mockMarket}
          image={mockImage}
          toggleFavorite={mockToggleFavorite}
          controlFavorites={mockControlFavorites}
        />)

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('mapDispatchToProps', () => {
    let mockMarket;
    let mockDispatch;

    beforeEach(() => {
      mockMarket = {
        id: '1234',
        marketname: 'Golden Farmers Markets',
        favorite: true
      }
      mockDispatch = jest.fn();
    });

    it('should call dispatch with toggleFavorite', () => {
      const actionToDispatch = toggleFavorite(mockMarket.id, mockMarket.favorite);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.toggleFavorite(mockMarket.id, mockMarket.favorite);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch with controlFavorites', () => {
      const actionToDispatch = controlFavorites(mockMarket);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.controlFavorites(mockMarket);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

  });
});
