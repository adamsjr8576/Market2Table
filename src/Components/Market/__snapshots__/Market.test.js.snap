// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Market Market Component should match the snapshot with all the correct data passed in 1`] = `
<article
  className="market-article"
>
  <Link
    className="market-link-to-page"
    to="/markets/1234"
  >
    <h1
      className="market-article-name"
    >
      Golden Farmers Markets
    </h1>
  </Link>
  <button
    className="favorites-button"
    onClick={[Function]}
    type="button"
  >
    <img
      alt="chili icon"
      className="chili-icon"
      src="./redChili.png"
    />
  </button>
</article>
`;
